pkg/clusterversion/cockroach_versions.go

```diff
@@ -518,21 +521,25 @@ var versionsSingleton = keyedVersions{
 		Version: roachpb.Version{Major: 21, Minor: 2, Internal: 16},
 	},
 	{
-		Key:     PublicSchemasWithDescriptors,
+		Key:     InsertPublicSchemaNamespaceEntryOnRestore,
```

Comment:
@ajwerner not sure if this is okay, I put this ahead since we want the migration keyed on `InsertPublicSchemaNamespaceEntryOnRestore` to go before `PublicSchemasWithDescriptors`. Figured it might be okay since v22 is not released at all yet, however playing it safe and just rekeying `PublicSchemasWithDescriptor` after `InsertPublicSchemaNamespaceEntryOnRestore` without rearranging the other versions might be better